deploy/jbossas/modeshape-jbossas-subsystem/src/main/java/org/modeshape/jboss/subsystem/ModeShapeSubsystemXMLReader_3_0.java

```diff
@@ -31,7 +31,7 @@
 import org.jboss.staxmapper.XMLElementReader;
 import org.jboss.staxmapper.XMLExtendedStreamReader;
 
-public class ModeShapeSubsystemXMLReader_2_1 implements XMLStreamConstants, XMLElementReader<List<ModelNode>> {
+public class ModeShapeSubsystemXMLReader_3_0 implements XMLStreamConstants, XMLElementReader<List<ModelNode>> {
```

Comment:
Will we need to keep a `ModeShapeSubsystemXMLReader_2_1` implementation around that reads the older configuration but builds the newer model? IIUC, that would allow us to parse 4.x configurations, and Wildfly would automatically write them out using the latest XML representation. (I know this is unrelated to this issue, but I would think that if we're going to want one then it'd be easier to start with it sooner than later.) Thoughts?